src/PlusDataCollection/OptiTrack/vtkPlusOptiTrack.cxx

```diff
@@ -313,21 +331,32 @@ PlusStatus vtkPlusOptiTrack::InternalCallback(sFrameOfMocapData* data)
 {
   LOG_TRACE("vtkPlusOptiTrack::InternalCallback");
 
-  this->Internal->MatchTrackedTools();
-
   const double unfilteredTimestamp = vtkPlusAccurateTimer::GetSystemTime();
 
-  sDataDescriptions* dataDescriptions;
-  this->Internal->NNClient->GetDataDescriptions(&dataDescriptions);
-
-  int numberOfRigidBodies = data->nRigidBodies;
-  sRigidBodyData* rigidBodies = data->RigidBodies;
+  if (this->FrameNumber == 0)
+  {
+    // do an initial match of tracked tools
+    this->Internal->MatchTrackedTools();
```

Comment:
Checking FrameNumber might not be that robust (if you disconnect and connect the device again then FrameNumber will not be reset, etc). It would be probably better to either ensure in InternalConnect that tracked tools are refreshed, or use a mechanism that relies for example on LastToolUpdateTime (if negative it means that there haven't been any updates ever; you could set LastToolUpdateTime to -1 in InternalConnect).